src/main/java/chess/domain/Board.java

```diff
@@ -0,0 +1,50 @@
+package chess.domain;
+
+import chess.domain.exception.IllegalPieceMoveException;
+import chess.domain.piece.EmptyPiece;
+import chess.domain.piece.Piece;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class Board {
+    private final Map<Position, Piece> piecePosition;
+
+    public Board(Map<Position, Piece> pieces) {
+        piecePosition = new HashMap<>(pieces);
```

Comment:
```suggestion
    private final HashMap<Position, Piece> piecePosition;

    public Board(HashMap<Position, Piece> pieces) {
        piecePosition = pieces;
```
보드는 해시맵을 사용하는 것으로 고정이라면 위와같이 변경하지 않은 이유가 있을까요???

Reply:
보드를 해쉬맵으로 사용하는 것을 고정하려던 의도는 없었습니다!!
다만 습관적으로 Map에 대한 구현체를 HashMap으로 만들었던 것 같아요 ㅎㅎ
Board의 필드를 Map이라는 자료형 적 성질만 가져가려 했기 떄문에
상세 구현타입인 HashMap으로 명시하지 않았던 것 같습니다 ㅎㅎ

Reply:
방어적 복사를 하려했습니다!!
생각해보니 방어적 복사를 HashMap 으로 한다면 결국 해당 필드는 HashMap 밖에 안 되겠군요!!
방어적 복사는 하고싶고 HashMap 으로 인한 로직이 생기는 것은  않고 싶고 참 고민이네요..
하지만 필드를 Map으로 유지하는 것이 좋아 보입니다!!
왜냐면 현재 저는 해당 필드를 Map 이상으로 보지 않았기 때문에 그 의도를 전달할 수 있을 뿐더러
후에 Map 구현체를 바꾼다면 코드 변경이 적지 않을까 생각이 듭니다!!